homeassistant/components/crownstone/config_flow.py

```diff
@@ -0,0 +1,299 @@
+
+        spheres = {sphere.name: sphere.cloud_id for sphere in manager.cloud.cloud_data}
+        usb_path = self.entry.options.get(CONF_USB_PATH)
+        usb_sphere = self.entry.options.get(CONF_USB_SPHERE)
+
+        options_schema = vol.Schema(
+            {vol.Optional(CONF_USE_USB_OPTION, default=usb_path is not None): bool}
+        )
+        if usb_path is not None and len(spheres) > 1:
+            options_schema = options_schema.extend(
+                {
+                    vol.Optional(
+                        CONF_USB_SPHERE_OPTION,
+                        default=manager.cloud.cloud_data.spheres[usb_sphere].name,
+                    ): vol.In(spheres.keys())
+                }
+            )
+
+        if user_input is not None:
+            if user_input[CONF_USE_USB_OPTION] and usb_path is None:
+                self.spheres = spheres
+                return await self.async_step_usb_config_option()
+            if not user_input[CONF_USE_USB_OPTION] and usb_path is not None:
+                self.updated_options[CONF_USB_PATH] = None
+                self.updated_options[CONF_USB_SPHERE] = None
+            elif (
+                CONF_USB_SPHERE_OPTION in user_input
+                and spheres[user_input[CONF_USB_SPHERE_OPTION]] != usb_sphere
+            ):
+                sphere_id = spheres[user_input[CONF_USB_SPHERE_OPTION]]
+                user_input[CONF_USB_SPHERE_OPTION] = sphere_id
+                self.updated_options[CONF_USB_SPHERE] = sphere_id
+
+            return self.async_create_entry(title="", data=self.updated_options)
+
+        return self.async_show_form(step_id="init", data_schema=options_schema)
+
+    async def async_step_usb_config_option(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Set up a Crownstone USB dongle."""
+        list_of_ports = await self.hass.async_add_executor_job(
+            serial.tools.list_ports.comports
+        )
+        ports_as_string = list_ports_as_str(list_of_ports, False)
+
+        if user_input is not None:
+            selection = user_input[CONF_USB_PATH]
+
+            if selection == MANUAL_PATH:
+                return await self.async_step_usb_manual_config_option()
+            if selection != REFRESH_LIST:
+                selected_port: ListPortInfo = list_of_ports[
+                    ports_as_string.index(selection)
+                ]
+                usb_path = await self.hass.async_add_executor_job(
+                    usb.get_serial_by_id, selected_port.device
+                )
+                self.updated_options[CONF_USB_PATH] = usb_path
+                return await self.async_step_usb_sphere_config_option()
+
+        return self.async_show_form(
+            step_id="usb_config_option",
+            data_schema=vol.Schema(
+                {vol.Required(CONF_USB_PATH): vol.In(ports_as_string)}
+            ),
+        )
+
+    async def async_step_usb_manual_config_option(
```

Comment:
How would you suggest we do this? Since most of the code in the functions in used to drive the flow itself. Perhaps a base class that both `OptionsFlow` and `ConfigFlow` derive from?

Reply:
Check how we inherit flows in the zwave_js integration config flow module. Something like that I think could work.